src/wallet/wallet.cpp

```diff
@@ -988,9 +985,7 @@ bool CWallet::AddToWallet(const CWalletTx& wtxIn, bool fFlushOnClose)
 bool CWallet::LoadToWallet(const CWalletTx& wtxIn)
 {
     uint256 hash = wtxIn.GetHash();
-
-    mapWallet[hash] = wtxIn;
-    CWalletTx& wtx = mapWallet[hash];
+    CWalletTx& wtx = mapWallet[hash] = wtxIn;
```

Comment:
I think this isn't correct: C syntax `A = B = C = D = X` sets A,B,C,D to X.. So it assigns `wtx` to a reference to `wtxIn`, not to one to `mapWallet[hash]` as before.

Reply:
@laanwj the following:
```
#include <map>
#include <string>
#include <iostream>
using namespace std;
int main() {
  map<int,string> m;
  string s = "hello";
  string& r = m[1] = s;
  r.append(" world");
  cout << s << endl;
  cout << r << endl;
  cout << m[1] << endl;
  return 0;
}
```
gives:
```
hello
hello world
hello world
```
So I think it's correct. I can make it more explicit and still avoid the 2nd lookup.

Reply:
Yes you are correct. Assignment happens from right to left. However I am not really a fan of the multiple assignment per line in this case since we are mixing reverence and value type assigmnents. It's a personal thing I guess but I usually like to separate such logic in multi-line as it helps with debugging etc. The compiler will optimize it anyway :-)

Reply:
Done.

Reply:
Good to know, thanks, didn't realize that, but yes it's better to avoid it to avoid confusion.